### ISSUE TITLE
Character count fails if ID starts with a number.

### ISSUE THREAD

**Issue by josef-vlach** [alphagov/govuk-frontend#2058]
## Description of the issue
When character count is used on a textarea with `id` which start with a number, it fails with an exception:

```
all.js:1501 Uncaught DOMException: Failed to execute 'querySelector' on 'Element': '[id=1_details-info]' is not a valid selector.
    at new CharacterCount (http://localhost/submissions/assets/lib/govuk-frontend/govuk/all.js:1501:34)
    at http://localhost/submissions/assets/lib/govuk-frontend/govuk/all.js:2515:5
    at NodeList.forEach (<anonymous>)
    at nodeListForEach (http://localhost/submissions/assets/lib/govuk-frontend/govuk/all.js:14:18)
    at Object.initAll (http://localhost/submissions/assets/lib/govuk-frontend/govuk/all.js:2514:3)
    at http://localhost/submissions/form/test/page?se=t&ff=t:1012:34
```

## Steps to reproduce the issue
Having a textarea with `id` which start with a number:
```
<textarea
    class="govuk-textarea govuk-js-character-count govuk-input--width-30"
    id="1_details"
    name="1_details"
    rows="5"
    aria-describedby="1_details-info"></textarea>
 ```

## Current behaviour
`querySelector` in character-count.js crashes when `id` starts with a number:

https://github.com/alphagov/govuk-frontend/blob/9b320cece2177f3a9749d8150e250fe8978e7eef/src/govuk/components/character-count/character-count.js#L8-L10

(This is a minimal reproduction of the issue)
```
> document.querySelector('[id=1_details-info]');
VM711:1 Uncaught DOMException: Failed to execute 'querySelector' on 'Document': '[id=1_details-info]' is not a valid selector.
    at <anonymous>:1:10
```

The reason for the crash is that `id` selector is not being quoted.

## Expected behaviour
`querySelector` works when `id` starts with a number:

```
> document.querySelector('[id="1_details-info"]');
<div id=​"1_details-info" class=​"govuk-hint govuk-character-count__message" aria-live=​"polite">
  You can enter up to 1000 characters
​</div>
```
It works when `id` selector is quoted: `id="1_details-info"`.

**Comment by SandipKhedekar** [comment]
As the above issue has been fixed long back. May we please know when it will be released?

**Comment by 36degrees** [comment]
@SandipKhedekar this was fixed in [v3.11.0](https://github.com/alphagov/govuk-frontend/releases/tag/v3.11.0) which has just been released this afternoon.
